engine/src/main/java/com/datatorrent/stram/cli/ApexCli.java

```diff
@@ -3516,6 +3517,16 @@ private void checkPlatformCompatible(AppPackage ap)
     }
   }
 
+  private void checkJavaVersionCompatible(AppPackage ap)
+  {
+    String buildJdkVersion = ap.getBuildJdkVersion();
+    String systemVersion = System.getProperty("java.version");
+    if (VersionInfo.compare(buildJdkVersion, systemVersion) > 0) {
```

Comment:
@oliverwnk @vrozov In that case the only way to implement this check is to check each and every class file in the APA (i.e. each class in each jar of the APA) and compare the class version against that the current Java supports. If there is anything newer then generate an error